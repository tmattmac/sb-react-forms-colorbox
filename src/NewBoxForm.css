.NewBoxForm {
    padding: 0.5em;
    background-color: #dddddd;
    border-radius: 1em;
    display: inline-block;
}

.NewBoxForm label {
    display: block;
}

.NewBoxForm input {
    display: block;
    width: 250px;
    margin-bottom: 0.5em;
}