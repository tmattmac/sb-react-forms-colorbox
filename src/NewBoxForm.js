import React, { useState } from 'react';
import './NewBoxForm.css';

const INIT_STATE = {
    width: '',
    height: '',
    color: '#000000'
};

function NewBoxForm({ addBox }) {
    const [formData, setFormData] = useState(INIT_STATE);
    const handleSubmit = e => {
        e.preventDefault();
        addBox(formData);
        setFormData(INIT_STATE);
    }
    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(form => ({ ...form, [name]: value }));
    }

    return (
        <form onSubmit={handleSubmit} className="NewBoxForm" data-testid="form">
            <label htmlFor="width">Width: </label>
            <input
                id="width"
                name="width"
                value={formData.width}
                type="number"
                min="1"
                onChange={handleChange}
                required
            />
            <label htmlFor="height">Height: </label>
            <input
                id="height"
                name="height"
                value={formData.height}
                type="number"
                min="1"
                onChange={handleChange}
                required
            />
            <label htmlFor="color">Color: </label>
            <input
                id="color"
                name="color"
                value={formData.color}
                type="color"
                onChange={handleChange}
            />
            <button>Add Box</button>
        </form>
    );
}

export default NewBoxForm;
