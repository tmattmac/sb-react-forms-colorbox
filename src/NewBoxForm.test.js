import { render, screen } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('renders', () => {
  render(<NewBoxForm />);
});

test('matches last snapshot', () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});