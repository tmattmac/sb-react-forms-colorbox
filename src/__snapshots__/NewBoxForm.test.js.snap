// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
    data-testid="form"
  >
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      min="1"
      name="width"
      required=""
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      min="1"
      name="height"
      required=""
      type="number"
      value=""
    />
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value="#000000"
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
